05 create expense item

saemorris opened the pull request:
For issue #5 Create an Expense Item

This doesn't have the expenseItem added to a claim yet since that functionality has not yet been implemented.  

A test for opening the camera is partially complete.  When the imageButton is clicked, control is transferred to the camera app, causing the test to never end.  If anyone has any ideas on how to get around this, that would be much appreciated.

ItsTristan commented on ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/ExpenseItemActivity.java:
```diff
@@ -0,0 +1,245 @@
+ * Displays activity_create_expense_item activity, to give the user an 
+ * interface to add the name, date, category, amount spent, currency, 
+ * description and a photo of a receipt for expense items. 
+ */
```
This needs the license header

saemorris commented:
Whoops missed that, it is there now.

ItsTristan commented on ShinyExpenseTrackerTests/src/ca/ualberta/cs/shinyexpensetracker/test/ExpenseItemActivityTest.java:
```diff
@@ -0,0 +1,167 @@
+				 assertTrue("currnency != CAD", Currency.CAD == expense.getCurrency());
+				 assertTrue("description != description", "description" == expense.getDescription());
+				 assertTrue("bitmap != bitmap", bitmap == expense.getReceiptPhoto());
+			}
```
* Can you change `assertTrue` to `assertEquals`? The "==" operator isn't reliable when comparing objects.
* Maybe also add one or two, things like: `assertNotSame( "something definitely not the value of expense.getName()", expense.getName() );` to make sure we aren't getting false positives.
* Can you add a test for `ExpenseItemActivity.createExpenseItem()`? ie. when I hit done, I want to be sure that everything I just entered into the UI is actually saved into the object.

Otherwise, looks good. :+1:

saemorris commented:
* I've changed the asserts and added some more. 

* As for the last comment about testing createExpenseItem(), this will have to wait until we can actually save an expenseItem.  Then we can load in the save expenseItem and check to make sure that it has the right information.  I've left a failing test for this so we don't forget to fix it in the future.
